docs/sdkx-cocos2dx/migration-guide.mdx

```diff
@@ -0,0 +1,635 @@
+| ---------------------------- | --------------------------------------------------------------------------------------------- | --------------------------------------------------------------------- |
+| Register device token        | `Core.registerDeviceToken(context, regid);`                                                   | `HelpshiftCocosBridge.registerPushToken ("fcm-token");`               |
+| Request unread message count | `HelpshiftCocos2dx::getNotificationCount(true/false, didReceiveNotificationCountListenerArg)` | `HelpshiftCocos2dx::requestUnreadMessageCount(shouldFetchFromServer)` |
+
+The delegate method that receives the unread message count has also changed, Legacy SDK Code -
+
+```cpp
+
+void didReceiveNotificationCountListenerArg(int count) {
+    CCLOG("HelpshiftCocos2dx::didReceiveNotificationCountListenerArg : %d.", count);
+}
+// Get count synchronously from cache.
+HelpshiftCocos2dx::getNotificationCount(false, didReceiveNotificationCountListenerArg);
+// OR
+// asynchronously from the server.
+HelpshiftCocos2dx::getNotificationCount(true, didReceiveNotificationCountListenerArg);
+
+```
+
+SDK X code -
+
+```cpp
+void getUnreadMessageCount(Ref *pSender) {
+  bool shouldFetchFromServer = true;
+  HelpshiftCocos2dx::requestUnreadMessageCount(shouldFetchFromServer);
+}
+
+void updateBadgeCount(int count) {
+  // Your UI Logic here
+}
+
+void handleHelpshiftEvent(const char *eventName, cocos2d::ValueMap &eventData) {
+    if (strcmp(eventName, RECEIVED_UNREAD_MESSAGE_COUNT) == 0) {
+        if (eventData.find(DATA_MESSAGE_COUNT) != eventData.end()) {
+          int count = eventData[DATA_MESSAGE_COUNT].asInt();
+          bool countFromCache = eventData[DATA_MESSAGE_COUNT_FROM_CACHE].asBool();
+          // Use the 'count' variable for further processing
+          updateBadgeCount(count);
+      }
+    }
+}
+```
+
+Read more about unread message count delegate [here](/sdkx-cocos2dx/delegates/#unread-message-count).
+
+#### Handle notification
+
+To handle notification clicks, replace legacy SDK code -
+
+```java
+@Override
+public void onMessageReceived(RemoteMessage message) {
+  Map<String, String> data = message.getData();
+  String origin = data.get("origin");
+  if (origin != null && origin.equals("helpshift")) {
+    Core.init(Support.getInstance());
+    try {
+        Core.install(getApplication(),
+                    <API_KEY>,
+                    <DOMAIN_NAME>,
+                    <APP_ID>,
+                    config);
+      } catch (InstallException e) {
+        // install credentials are incorrect
+      }
+    Core.handlePush(this, data);
+  }
+}
+```
+
+with SDK X code -
+
+```cpp
+@Override
+public void onMessageReceived(RemoteMessage message) {
+  Map<String, String> data = message.getData();
+  String origin = data.get("origin");
+  if (origin != null && origin.equals("helpshift")) {
+    HelpshiftCocosBridge.handlePushNotification(getApplicationContext(), data);
```

Comment:
there should be install call first here i think

Reply:
We do a `install` call with cached values inside `handlePushNotification` method.